week2/assignment/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"sync"
+)
+
+func readFileByLine(filePath string, lines chan<-string, processingFile <-chan string, wg *sync.WaitGroup) {	
+	file, err := os.Open(filePath)
+	
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer func() {
+		file.Close();
+		
+		<- processingFile
+		if len(processingFile) == 0 {
+			close(lines)
```

Comment:
I think it's not guaranteed that only one goroutine would run this line of code.